src/main/java/org/aksw/simba/lemming/creation/Equivalent.java

```diff
@@ -0,0 +1,180 @@
+package org.aksw.simba.lemming.creation;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.jena.ontology.ConversionException;
+import org.apache.jena.ontology.OntClass;
+import org.apache.jena.ontology.OntProperty;
+import org.apache.jena.ontology.OntResource;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.rdf.model.Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Represents an OntClass or an OntProperty and its equivalent classes or
+ * properties, respectively.
+ *
+ * @param <T> OntProperty, OntClass
+ */
+public class Equivalent<T extends OntResource> {
+	private static final Logger LOGGER = LoggerFactory.getLogger(Equivalent.class);
+
+	/**
+	 * single name to characterize all its equivalents
+	 */
+	private String name;
+
+	/**
+	 * should be instanceof OntProperty or OntClass
+	 */
+	private T attribute;
+
+	/**
+	 * URIs of all equivalents
+	 */
+	private Set<String> equivalents = new HashSet<>();
+
+	public Equivalent(T current) {
+		super();
+		this.name = ((Resource) current).getURI();
+		this.attribute = current;
+		this.equivalents.add(name);
+	}
+
+	/**
+	 * This method adds the URI to the this.equivalents set and updates the object's
+	 * own name. If the object is an OntProperty, the domain and range information
+	 * is also added to the class' object
+	 * 
+	 * @param current OntProperty / OntClass object
+	 */
+	public void addEquivalent(T current) {
+		String uri = ((Resource) current).getURI();
+		if (uri != null) {
+			this.equivalents.add(uri);
+			setName(compareNames(uri, name));
+
+			if (current instanceof OntProperty) {
+				List<? extends OntResource> domainList = ((OntProperty) current).listDomain().toList();
```

Comment:
Defining something like
```java
OntProperty currentProperty = current.asProperty();
```
would make the following lines much easier (i.e., remove all the single casts)